src/dbnode/storage/README.md

```diff
@@ -4,17 +4,41 @@ Storage related documentation.
 
 ## Flush consistency model
 
-Flush occurs in the following steps:
+Warm and cold flush each have their own independent lifecycle:
+
+Warm flush:
+  - warm flush cleanup
+    - expired/duplicate index files
+    - inactive snapshot/namespace files.
+    - index/data snapshots
   - data warm flush
   - rotate commit log
+  - data and index snapshot
+    - drops rotated commit log when we are done
+  - index warm flush
+
+Cold flush:
+  - cold flush cleanup
+    - out of retention/compacted data files
+    - data for no longer owned shards
   - data cold flush
     - rotate cold mutable index segments
     - flush cold tsdb data and write most files to disk (except checkpoint files)
     - flush cold index data to disk and reload
```

Comment:
For my own edification, why do warm and cold flushes flush their data and indices in the opposite order?